### PR TITLE
Initialize the Windows COM library on all threads, ensure Windows DDE conversations are finished in `open_link`

### PR DESCRIPTION
**Pull request by Robyt3**
## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
Why did success change from > 32 to != 0?

**Comment by Robyt3** [comment]
`ShellExecuteExW`:

> Returns TRUE if successful; otherwise, FALSE. Call GetLastError for extended error information.

`ShellExecuteW`:

> If the function succeeds, it returns a value greater than 32. If the function fails, it returns an error value that indicates the cause of the failure. The return value is cast as an HINSTANCE for backward compatibility with 16-bit Windows applications. It is not a true HINSTANCE, however. It can be cast only to an INT_PTR and compared to either 32 or the following error codes below.

Consistency isn't really a strength of the Windows API.

File: src/base/system.h [hunk 1]
```diff
@@ -2563,11 +2563,13 @@ class CCmdlineFix
 /**
  * This is a RAII wrapper to initialize/uninitialize the Windows COM library,
  * which may be necessary for using the open_file and open_link functions.
+ * Must be used on every thread. Pass true to the constructor on threads
+ * that own a window (i.e. pump a message queue).
```
Comment (heinrich5991) [inline src/base/system.h:2567]: That comment isn't really true, is it? It's automatically called on every thread other than the main thread.

Comment (Robyt3) [reply src/base/system.h:2567]: True. How about the following?

> Must be used on every thread. It's automatically used on threads created with thread_init.
> Pass true to the constructor on threads that own a window (i.e. pump a message queue).

Comment (heinrich5991) [reply src/base/system.h:2567]: Sounds good.

**Pull request merged**